### DISCUSSION TITLE
Keycloak Admin Client - ID Token

### DISCUSSION THREAD

**Discussion by a-maggioni** [keycloak/keycloak#8668] [Q&A]
I would like to use the keycloak admin client to get an ID token programmatically.
I've tried to use the `getAccessToken` method of `TokenManager` class, but the response contains only an access token (without ID token). As far as I know his is because the `grantToken` method doesn't add the `openid` scope in the request.
Is there an alternative way to get an ID token?

**Comment by stianst** [comment]
The purpose of admin client is to invoke the Admin APIs (OAuth), and not as a library to authenticate with (OIDC). So, it sounds like you should be looking for a different library.
